FlightSQL example is broken

**Describe the bug**
<!--
A clear and concise description of what the bug is.
-->

FlightSQL example (`arrow-flight/examples/flight_sql_server.rs`) is broken. Running `cargo test -p arrow-flight  --features=flight-sql-experimental --examples` got compilation error:

```
error[E0433]: failed to resolve: use of undeclared crate or module `fs`
   --> arrow-flight/examples/flight_sql_server.rs:504:17
    |                                                                                                              
504 |         let _ = fs::remove_file(path.clone());
    |                 ^^ use of undeclared crate or module `fs`  
                                                                                                                                                                                                                                       
error[E0412]: cannot find type `FlightSqlServiceClient` in this scope                                              
   --> arrow-flight/examples/flight_sql_server.rs:490:47                                                           
    |                                                                                                              
490 |     async fn client_with_uds(path: String) -> FlightSqlServiceClient {                                       
    |                                               ^^^^^^^^^^^^^^^^^^^^^^
    |                               
...
```

**To Reproduce**
<!--
Steps to reproduce the behavior:
-->

**Expected behavior**
<!--
A clear and concise description of what you expected to happen.
-->

```
cargo test -p arrow-flight  --features=flight-sql-experimental --examples

   Compiling arrow-flight v29.0.0 (/arrow-rs/arrow-flight)
    Finished test [unoptimized + debuginfo] target(s) in 1.76s
     Running unittests examples/flight_sql_server.rs (target/debug/examples/flight_sql_server-609b62f910a8643f)

running 1 test
test tests::test_select_1 ... ok
```

**Additional context**
<!--
Add any other context about the problem here.
-->

--- a/.github/workflows/arrow_flight.yml
+++ b/.github/workflows/arrow_flight.yml
@@ -56,6 +56,9 @@ jobs:
       - name: Test --all-features
         run: |
           cargo test -p arrow-flight --all-features
+      - name: Test --examples
+        run: |
+          cargo test -p arrow-flight  --features=flight-sql-experimental --examples
       - name: Verify workspace clean
         run: git diff --exit-code
 

--- a/arrow-flight/examples/flight_sql_server.rs
+++ b/arrow-flight/examples/flight_sql_server.rs
@@ -482,8 +482,14 @@ impl ProstMessageExt for FetchResults {
 mod tests {
     use super::*;
     use futures::TryStreamExt;
+    use std::fs;
+    use tempfile::NamedTempFile;
+    use tokio::net::{UnixListener, UnixStream};
+    use tokio_stream::wrappers::UnixListenerStream;
+    use tonic::transport::Endpoint;
 
     use arrow::util::pretty::pretty_format_batches;
+    use arrow_flight::sql::client::FlightSqlServiceClient;
     use arrow_flight::utils::flight_data_to_batches;
     use tower::service_fn;
 

